### DISCUSSION TITLE
Support system ROMs from C64 Forever and VICE

### DISCUSSION THREAD

**Discussion by kjetil-f** [markus-zzz/myc64-pocket#25] [Ideas]
I'm not i'm not criticizing the use of zimmers.nets files, but it's always nice to have a legal alternative. C64 Forever and VICE are considered legal, it would be nice if these would work with the core at some point.

**Comment by markus-zzz** [comment]
Sure. Is there any indication that they do not work?
Where can they be found / how does one get hold of them?

**Reply by kjetil-f** [reply]
Actually, they might work. I have not been able to test it myself, because of some technical issues on my end. Also, I have next to no knowledge on this subject, but based on file names and file sizes here is what I found out:

In C64 Forever I can find "c-64-basic.rom" and "c-64-kernal.rom", which I'm guessing could be the same as the files in VICE called "basic-901226-01.bin" and "kernal-901227-01.bin"

I can also found "c-64-chargen.rom" in C64 Forever and "chargen-901225-01.bin" in VICE. (Guessing that's the character ROM)

All also found "dos1540-325302+3-01.bin" and "dos1541-325302-01+901229-05.bin" in VICE, but only "c-drive-1541.rom" in C64 Forever.

**Reply by dukestah** [reply]
I don't get the point. All of the mentioned firmware files included in the VICE package are available at zimmers.net and they are the very same bit-wise. So getting them via VICE installation won't make them more legal compared to get them from zimmers.net.
To my knowledge, there is a sort of freeware kernal rom, developed by some people in the C65 universe.

**Reply by kjetil-f** [reply]
That's a good question. And functionally they are the same. Most people won't care, but it's just for having a "legal" alternative (or an alternative in general). Which is nice in the emulation/FPGA scene. Basically, it doesn't matter if the files are identical. But where you source them from will depend if they are considered legal or not.

Since VICE are used in commercial products like C64 Forever and THEC64 Mini/Maxi one can assume they are in the clear, legally speaking. But then again, I think the copyrights with C64 ROMs are a bit unclear, depending on who you ask.

**Comment by markus-zzz** [comment]
I would expect them to work.
Let us know once you managed to test and then I can update the project README with some notes on this.

The 1541 is a bit more picky though since there were different variants of the drive and you need to match the exact model.

**Reply by kjetil-f** [reply]
I will do that when when time allows.
